### DISCUSSION TITLE
wireguard config

### DISCUSSION THREAD

**Discussion by trinitygirl** [worried-networking/yandex-cloud-vpn#2] [General]
I am not good at wireguard, but trying to set up a vpn through your way. Your script works, but I am struggling after step 9. Can you add a bit more details? How do I configure wireguard and/or what exactly should I do after executing do.sh?

**Comment by worried-networking** [comment]
@trinitygirl where would you like to use the VPN? On what OS?

The script only configures the server side and displays the config you need to add to the client.

**Reply by trinitygirl** [reply]
Tried Ubuntu 22, now trying on macOS. Seems like I have troubles configuring WG on Ubuntu.

**Reply by worried-networking** [reply]
For Ubuntu you just need to save the config provided by the script to `/etc/wireguard/wg0.conf` and execute `sudo wg-quick up wg0` — it should prepare everything required.

For macOS it's all done via the GUI. Just add the config there, and you're done.

**Reply by trinitygirl** [reply]
wg show shows nothing - is it should be like this?

**Reply by worried-networking** [reply]
No. macOS of Ubuntu?

**Reply by trinitygirl** [reply]
Ubuntu, macOS configuring right now, havn't tested yet

**Reply by worried-networking** [reply]
What was the output of `wg-quick`?

**Reply by worried-networking** [reply]
I mean `sudo wg-quick up wg0` — that command should've configured the interface and establish the connection for you.

**Reply by trinitygirl** [reply]
sudo wg-quick up wg0
[#] ip link add wg0 type wireguard
[#] wg setconf wg0 /dev/fd/63
[#] ip -4 address add 10.128.0.27/24 dev wg0
[#] ip -6 address add fd0d:86fa:c3bc::1/64 dev wg0
RTNETLINK answers: Permission denied
[#] ip link delete dev wg0

**Reply by worried-networking** [reply]
I'm guessing you have ipv6 disabled. Can you try replacing `AllowedIPs = 0.0.0.0/0, ::/0` with `AllowedIPs = 0.0.0.0/0` in the config?

**Reply by worried-networking** [reply]
BTW, I'm surprised by the listed IPs. Have you changed the default values? You should've had `192.168.55.2` by default for the first client.

**Reply by trinitygirl** [reply]
I did change address at some point, I guess I need to start it from scratch. And yes, now I see I didn't allowed IPv6

**Reply by trinitygirl** [reply]
Looks like I finally don't get errors

sudo wg-quick up wg0
[#] ip link add wg0 type wireguard
[#] wg setconf wg0 /dev/fd/63
[#] ip -4 address add 192.168.55.2 dev wg0
[#] ip link set mtu 1420 up dev wg0
[#] resolvconf -a wg0 -m 0 -x


Still don't understand how to use vpn ))) Did I miss something?

**Reply by worried-networking** [reply]
I'm not sure, to be honest :) Would you mind sharing the client config you're using? Do not forget to mask the endpoint and the private key.

From my understanding, `wg-quick` should've added a new default route. I haven't tried running the client on Ubuntu and not sure how exactly it was supposed to work :)

**Reply by trinitygirl** [reply]
I think I don't really understand how yandex cloud works. I have VM with 2 IPs. One of them should be used in WG configuration, shouldn't it?

**Reply by worried-networking** [reply]
That's correct. The VM have an internal IP and an external one. The external IP also should've been displayed by the `do.sh` script after the VM is created.

**Reply by trinitygirl** [reply]
look what I've got

sudo wg show

interface: wg0
  public key: hG5YnrStDI5p732WaGXh2MI38Gw4GY8rRhDqcU1V4Vw=
  private key: (hidden)
  listening port: 51820

And then?

**Reply by worried-networking** [reply]
I have no idea, sorry. It just works for me out of the box. I have the following output on the client:

```
interface: wg0
  public key: tQ2dgoBCvZcwVVwEV3xJRiO0R2jxTfrnlZ7HK5I22Xo=
  private key: (hidden)
  listening port: 28175

peer: i2xQ/wDgiJBHw0uTN4l3nc7Vk9UdjcbiOIedaRiIIG0=
  endpoint: <remote ip>:28175
  allowed ips: 192.168.2.0/24
  latest handshake: 26 seconds ago
  transfer: 768.49 MiB received, 280.21 MiB sent
  persistent keepalive: every 30 seconds
```

And this one from the server:

```
interface: wg0
  public key: i2xQ/wDgiJBHw0uTN4l3nc7Vk9UdjcbiOIedaRiIIG0=
  private key: (hidden)
  listening port: 28175

peer: tQ2dgoBCvZcwVVwEV3xJRiO0R2jxTfrnlZ7HK5I22Xo=
  preshared key: (hidden)
  endpoint: <client ip>:28175
  allowed ips: 192.168.2.2/32
  latest handshake: 1 minute, 22 seconds ago
  transfer: 452.17 MiB received, 1.34 GiB sent
  persistent keepalive: every 30 seconds
```

**Reply by worried-networking** [reply]
Having `DNS = 8.8.8.8` and `AllowedIPs = 0.0.0.0/0` in the client config should've routed all your traffic via the tunnel. If it doesn't work — you'll have to investigate it.

**Reply by trinitygirl** [reply]
Thanks, I will figure this out. Thanks for your replies.

BTW what type of OS are you using?

**Reply by worried-networking** [reply]
I have tried Yandex.Cloud-based VPN on iOS and macOS — everything worked like a charm. Also, I have WG clients running on Ubuntu and OpenWRT, but I have configured those without the script.
